### PR TITLE
Cumulus 1372: Fix rules triggered by SF events

### PR DESCRIPTION
**Pull request by markdboyd**
**Summary:** Summary of changes

Addresses [CUMULUS-1372: Restrict sfSemaphoreDown to run only for the deployed stack](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1372)

## Changes

* Add Kes logic and update CF template so that Cloudwatch rules triggered by SF events only run for Step Functions in the current deployment
* Update env var for `backgroundQueueName` to include stack name

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing

### CODE REVIEW

File: packages/deployment/lib/kes.js [hunk 1]
```diff
@@ -320,6 +323,78 @@ class UpdatedKes extends Kes {
...
+
+      const initialPatternLength = JSON.stringify(rule.eventPattern).length;
+      // Pessimistically assume longest possible state machine ARN:
+      //    80 = max state machine length
+      //    64 = length of other ARN characters
+      //    2 = two double quotes
+      const arnLength = 80 + 64 + 2;
+      // Determine how many state machines can be added as conditions to the eventPattern
+      // before it would exceed the maximum allowed length of 2048 characters.
+      const stateMachinesPerRule = Math.ceil((2048 - initialPatternLength) / arnLength);
```
Comment (Jkovarik) [inline packages/deployment/lib/kes.js]: 👍

File: packages/deployment/lib/kes.js [hunk 2]
```diff
@@ -320,6 +323,78 @@ class UpdatedKes extends Kes {
...
+        const stateMachineName = `${prefixNoDash}${sfName}StateMachine`;
+        const stateMachineArnRef = `\$\{${stateMachineName}\}`;
+
+        newRule.stateMachines.push(stateMachineName);
+        newRule.eventPattern.detail.stateMachineArn.push(stateMachineArnRef);
+
+        updatedRules[`${ruleName}${ruleNumber}`] = newRule;
+      });
+
+      delete updatedRules[ruleName];
```
Comment (yjpa7145) [inline packages/deployment/lib/kes.js]: In what circumstance would `updatedRules[ruleName` exist at this point?

Comment (markdboyd) [reply packages/deployment/lib/kes.js]: Good point

File: packages/deployment/app/cloudformation.template.yml [hunk 3]
```diff
@@ -239,45 +239,6 @@ Resources:
   # SNS config END
   #################################################
 
-  #################################################
```
[the remark this answers is not in the document]
Comment (markdboyd) [reply packages/deployment/app/cloudformation.template.yml]: Yeah, moving this around just made it easier to debug the Cloudformation output file

**Pull request merged**